code/client/src/pages/Show/Swap.jsx

```diff
@@ -0,0 +1,268 @@
+import { TallRow } from '../../components/Grid'
+import { Col, Typography, Select, Image, Button, message } from 'antd'
+import React, { useCallback, useEffect, useState } from 'react'
+import { useSelector } from 'react-redux'
+import { mockCryptos } from './mock-cryptos'
+import ONEConstants from '../../../../lib/constants'
+import util from '../../util'
+import { DefaultTrackedERC20, HarmonyONE, withKeys } from '../../components/TokenAssets'
+import api from '../../api'
+import { InputBox } from '../../components/Text'
+import { trim } from 'lodash'
+const { Text } = Typography
+
+// TODO: some token's images are not available, we may want a CDN or other service that can retrieve icons dynamically.
+const cryptoIconUrl = (symbol) => `https://qokka-public.s3-us-west-1.amazonaws.com/crypto-logos/${symbol.toLowerCase()}.png`
+
+const TokenLabel = ({ token }) => (
+  <>
+    <Image
+      style={{ display: 'inline', paddingRight: '5px' }}
```

Comment:
image is distorted because of right padding

Reply:
Good catch, will fix it.